Updated AWS CFTs across all templates

ryanb-vega opened the pull request:
- CFT Parameter VegaExternalIdParameter: modified template to construct the ExternalID (vega:)
- updated usages of `VegaExternalIdParameter` to use the !Sub function to construct the string
- CFT Parameter CurBucketsParameter: modified template to construct both Bucket Resource ARN permissions rather than requiring the user to duplicate inputs and type out the ARN
- updated usages of `CurBucketsParameter` to create both required Bucket Resource ARN permissions for the MPA `S3SyncPolicy`
- IAM Role Trust Policies: use a `-` list for consistency across the templates
- all Parameter descriptions now use multi-line strings for improved readability

ryanb-vega committed:
Updated AWS CFTs across all templates
- CFT Parameter VegaExternalIdParameter: modified template to construct the ExternalID (vega:)
- updated usages of `VegaExternalIdParameter` to use the !Sub function to construct the string
- CFT Parameter CurBucketsParameter: modified template to construct both Bucket Resource ARN permissions rather than requiring the user to duplicate inputs and type out the ARN
- updated usages of `CurBucketsParameter` to create both required Bucket Resource ARN permissions for the MPA `S3SyncPolicy`
- IAM Role Trust Policies: use a `-` list for consistency across the templates
- all Parameter descriptions now use multi-line strings for improved readability

DanR-Vega reviewed (changes requested):
Ryan,
These updates look good, but require an update before approval.  Status' below:

vega-cft-mpa-voperate_sku.yaml
- Good deploy, verified bucket change and ExternalID change. OK!

vega-cft-mpa-vinform_voptimize_sku.yaml
- Good deploy, verified bucket change and ExternalID change. OK!

vega-cft-linked-vinform_voptimize_sku.yaml
- Good deploy, verified Roles and Policies look good.  OK!

vega-cft-linked-voperate_sku-all_actions.yaml
- Good deploy, verified Roles and Policies look good.  OK!

vega-cft-linked-voperate_sku-parking_actions_only.yaml
- Good deploy, verified Roles and Policies look good.  OK!

vega-cft-linked-voperate_sku-parking_tagging_actions_only.yaml
- Failed deploy, please remediate and I will test again!
- Reason:  TaggingPolicy - CREATE_FAILED
- Resource handler returned message: "Maximum policy size of 10240 bytes exceeded for role VegaOptimizer; ServiceLimitExceeded.    -Thanks! -Dan

DanR-Vega commented:
You can view or monkey with the test deploy in the Sandbox (2833) - https://us-west-2.console.aws.amazon.com/cloudformation/home?region=us-west-2#/stacks/stackinfo?filteringText=&filteringStatus=active&viewNested=true&stackId=arn%3Aaws%3Acloudformation%3Aus-west-2%3A283354751412%3Astack%2FDanLinkedOperateParkTagOnlytest%2Febb293b0-ac43-11ef-bba7-0a6b4439473f  - Thanks!

ryanb-vega committed:
- transitioned each IAM Policy from an inline policy to a managed policy for each example CFT
- split TaggingPolicy into groups based on the first letter of the service name to avoid IAM Policy length limitations
- updated automatically generated IAM Policy SIDs to a descriptive name

ryanb-vega committed:
added `is` to comments

Johnny-Vega reviewed (approved):
Reviewed and approved

ryanb-vega commented:
Converting this to a Draft PR until after the Change Freeze is over.
